### PR TITLE
fix: enforce max field length

### PR DESCRIPTION
**Pull request by fdlk**
issue: https://github.com/molgenis/molgenis-ui-form/issues/269

#### Checklist
- [x] Functionality works & meets specifications
- [x] Code reviewed
- [x] Code unit/integration/system tested
- [x] User documentation updated
- [x] Clean commits
- [x] No warnings during install
- [x] Updated flow typing

### CODE REVIEW

**Comment by marikaris** [comment]
I think we need to add an e2e test for this

File: src/components/field-types/SingleSelectFieldComponent.vue [hunk 1]
```diff
@@ -110,12 +110,14 @@
...
-        if (value) {
-          // Emit value changes to the parent (form)
-          this.$emit('input', value.id)
-        } else {
-          this.$emit('input', null)
-        }
+        this.fieldState.$dirty = true
+        this.fieldState.$pristine = false
+        this.fieldState.$touched = true
+        this.fieldState.$untouched = false
```
Comment (fdlk) [inline src/components/field-types/SingleSelectFieldComponent.vue]: @marikaris do we need these?

File: yarn.lock [hunk 2]
```diff
@@ -9230,9 +9230,9 @@ vue-router@^3.0.1:
   integrity sha512-vLLoY452L+JBpALMP5UHum9+7nzR9PeIBCghU9ZtJ1eWm6ieUI8Zb/DI3MYxH32bxkjzYV1LRjNv4qr8d+uX/w==
 
 vue-select@^2.6.3:
-  version "2.6.3"
-  resolved "https://registry.yarnpkg.com/vue-select/-/vue-select-2.6.3.tgz#c123630dbcda28225dd7697fc2f4f34333a5ae58"
-  integrity sha512-0zZ0/LEe9mNcDVKPiGSDun2qJ0e51ts5Z9f17XyL6bPMzw6rxNYPpx+fcSZsbfNYxwXuK3q4uFkucybw/EeQcA==
+  version "2.6.4"
```
Comment (fdlk) [inline yarn.lock]: @marikaris is this intentional?

**Comment by connoratrug** [comment]
is this a breaking change ? if so i suggest to update package, readme and changelog

File: test/e2e/specs/max-length-test.js [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+    browser.click('#validate-btn')
+    browser.expect.element('#max-length-example').to.have.attribute('class').which.contains('vf-form-invalid')
+    browser.expect.element('#max-length-string-example').to.have.attribute('class').which.contains('vf-invalid')
+    browser.expect.element('#max-length-text-example').to.have.attribute('class').which.contains('vf-invalid')
+    browser.expect.element('#max-length-url-example').to.have.attribute('class').which.contains('vf-invalid')
+    browser.expect.element('#max-length-script-example').to.have.attribute('class').which.contains('vf-invalid')
+    browser.expect.element('#max-length-email-example').to.have.attribute('class').which.contains('vf-invalid')
+    browser.end()
+  },
+  'String with not too many values should be valid': function (browser) {
```
Comment (connoratrug) [inline test/e2e/specs/max-length-test.js]: Non grammatical sentence, maybe something like:  String with length shorter then max length should be valid

**Comment by marikaris** [comment]
> When try to run it the example in dev mode and select the max length tab my browser locks and becomes unresponsive, i think we need to fix this before this can be merged

That's because you did not read my slack message in general. The script test had to be removed because it was not performant and it is unit tested anyway. The correct version is on my local repository because I was still working on it since the clearValue is still not working properly.

**Pull request merged**